catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java

```diff
@@ -258,22 +267,22 @@ public SourceResponse query(QueryRequest queryRequest) throws UnsupportedQueryEx
             LOGGER.debug("Received query: " + query);
         }
 
-        boolean canDoOpenSearch = setOpenSearchParameters(query, subject, restWebClient);
+        boolean canDoOpenSearch = setOpenSearchParameters(queryRequest, subject, restWebClient);
 
         if (canDoOpenSearch) {
 
             InputStream responseStream = performRequest(restWebClient);
 
-            response = new SourceResponseImpl(queryRequest, new ArrayList<Result>());
+            response = new SourceResponseImpl(queryRequest, new ArrayList<>());
 
             if (responseStream != null) {
                 response = processResponse(responseStream, queryRequest);
             }
         } else {
             if (StringUtils.isEmpty((String) metacardId)) {
-                OpenSearchFilterVisitor visitor = new OpenSearchFilterVisitor();
-                query.accept(visitor, null);
-                metacardId = visitor.getMetacardId();
+                openSearchFilterVisitor.resetState();
```

Comment:
Yeah, I don't think this code was making the best use of the opengis filter APIs.  I've refactored this to be more threadsafe